testing/test_hellsemble/hellsemble.py

```diff
@@ -177,25 +191,27 @@ def __fit_estimators_sequential(
                 list of fitted estimators and the list of masks indicating
                 which observations were used during fit of the estimators.
         """
+        X_train, X_val, y_train, y_val = train_test_split(
+            X, y, test_size=test_size, random_state=seed
+        )
         fitting_history: list[np.ndarray] = []
-        output_estimators = []
+        self.estimators = []
         coverage_counts = []
         performance_scores = []
-        failed_observations_idx = np.arange(X.shape[0])
-        X_fit, y_fit = X, y
+        failed_observations_idx = np.arange(X_train.shape[0])
+        X_fit, y_fit = X_train, y_train
```

Comment:
I'm not really sure if it works correctly without `.copy()`, unless reducing `X_fit` should be followed by reducing `X_train`